app/services/katello/pulp3/repository.rb

```diff
@@ -243,29 +250,33 @@ def ssl_remote_options
 
       def lookup_version(href)
         api.repository_versions_api.read(href) if href
-      rescue PulpcoreClient::ApiError => e
+      rescue api.api_exception_class => e
         Rails.logger.error "Exception when calling repository_versions_api->read: #{e}"
         nil
       end
 
       def lookup_publication(href)
         api.publications_api.read(href) if href
-      rescue PulpcoreClient::ApiError => e
+      rescue api.api_exception_class => e
         Rails.logger.error "Exception when calling publications_api->read: #{e}"
         nil
       end
 
       def remove_content(content_units)
-        data = PulpcoreClient::RepositoryVersionCreate.new(
-          remove_content_units: content_units.map(&:pulp_id))
-        api.repository_versions_api.create(repository_reference.repository_href, data)
+        if repo.root.content_type == "docker"
+          api.repositories_api.remove(repository_reference.repository_href, content_units: content_units.map(&:pulp_id))
+        else
+          api.repositories_api.modify(repository_reference.repository_href, remove_content_units: content_units.map(&:pulp_id))
+        end
       end
 
       def add_content(content_unit_href)
         content_unit_href = [content_unit_href] unless content_unit_href.is_a?(Array)
-        data = PulpcoreClient::RepositoryVersionCreate.new(
-            add_content_units: content_unit_href)
-        api.repository_versions_api.create(repository_reference.repository_href, data)
+        if repo.root.content_type == "docker"
+          api.repositories_api.add(repository_reference.repository_href, content_units: content_units_href)
+        else
+          api.repositories_api.modify(repository_reference.repository_href, add_content_units: content_unit_href)
+        end
```

Comment:
same 'override this in docker.rb' comments for both add_content and remove_content, although this almost either looks like a mismatch in bindings, or a pulp3 bug

Reply:
I have to use `add` here because `modify` is not an endpoint for `pulp_container`.  `modify` doesn't do dependency resolving, which the Pulp folks wanted to force on container content.